src/ops/queue.c

```diff
@@ -70,3 +158,384 @@ static void op_Q_N_set(const void *NOTUSED(data), scene_state_t *ss,
+    if (!ss->variables.q_grow && ss->variables.q_n < 1) {
+        ss->variables.q_n = 1;
+    }
+}
+
+static void op_Q_SUM_get(const void *NOTUSED(data), scene_state_t *ss,
+                         exec_state_t *NOTUSED(es), command_state_t *cs) {
+    int16_t *q = ss->variables.q;
+    int16_t q_n = ss->variables.q_n;
+    int16_t sum = 0;
+    for (int8_t i = 0; i < q_n; i++) { sum += q[i]; }
+    cs_push(cs, sum);
+}
+
+static void op_Q_MIN_get(const void *NOTUSED(data), scene_state_t *ss,
+                         exec_state_t *NOTUSED(es), command_state_t *cs) {
+    int16_t *q = ss->variables.q;
+    int16_t q_n = ss->variables.q_n;
+    int16_t min = INT16_MAX;
+    for (int8_t i = 0; i < q_n; i++) {
+        if (q[i] < min) { min = q[i]; }
+    }
+    cs_push(cs, min);
+}
+
+
+static void op_Q_MIN_set(const void *NOTUSED(data), scene_state_t *ss,
+                         exec_state_t *NOTUSED(es), command_state_t *cs) {
+    int16_t *q = ss->variables.q;
+    int16_t q_n = ss->variables.q_n;
+    int16_t min = cs_pop(cs);
+    for (int8_t i = 0; i < q_n; i++) {
+        if (q[i] < min) { q[i] = min; }
+    }
+}
+
+
+static void op_Q_MAX_get(const void *NOTUSED(data), scene_state_t *ss,
+                         exec_state_t *NOTUSED(es), command_state_t *cs) {
+    int16_t *q = ss->variables.q;
+    int16_t q_n = ss->variables.q_n;
+    int16_t max = INT16_MIN;
+    for (int8_t i = 0; i < q_n; i++) {
+        if (q[i] > max) { max = q[i]; }
+    }
+    cs_push(cs, max);
+}
+
+static void op_Q_MAX_set(const void *NOTUSED(data), scene_state_t *ss,
+                         exec_state_t *NOTUSED(es), command_state_t *cs) {
+    int16_t *q = ss->variables.q;
+    int16_t q_n = ss->variables.q_n;
+    int16_t max = cs_pop(cs);
+    for (int8_t i = 0; i < q_n; i++) {
+        if (q[i] > max) { q[i] = max; }
+    }
+}
+
+
+static void op_Q_RND_get(const void *NOTUSED(data), scene_state_t *ss,
+                         exec_state_t *NOTUSED(es), command_state_t *cs) {
+    int16_t *q = ss->variables.q;
+    int16_t q_n = ss->variables.q_n;
+    cs_push(cs, q[rand() % q_n]);
+}
+
+
+static void op_Q_RND_set(const void *NOTUSED(data), scene_state_t *ss,
+                         exec_state_t *NOTUSED(es), command_state_t *cs) {
+    int16_t *q = ss->variables.q;
+    int16_t q_n = ss->variables.q_n;
+    int16_t rnd = cs_pop(cs);
+    int16_t tmp;
+    int8_t a, b;
+
+    if (rnd > 0) {
+        // all elements random between 0 and rnd
+        for (int8_t i = 0; i < q_n; i++) { q[i] = rand() % rnd; }
+    }
+    else if (rnd < 0) {
+        // switch random elements rnd nb times
+        rnd = rnd < (-3 * q_n) ? (-3 * q_n) : rnd; // not more than 3*q_n times
+        for (int16_t i = rnd; i < 0; i++) {
+            a = rand() % q_n;
+            b = rand() % q_n;
+            tmp = q[a];
+            q[a] = q[b];
+            q[b] = tmp;
+        }
+    }
+    else {
+        // what to to with rnd = 0????
```

Comment:
old comment? the doc says it will do nothing

Reply:
Basically RND is doing a few things:

Q.RND // return random element
Q.RND x (x>0) // set elements to random value between 0 and x
Q.RND x (x<0) // swaps elements -x times 

The comment was to remind the reader (you?) that Q.RND 0 doesn't do anything.

I realize it's a but multi purpose, OTOH it's filling gaps that would otherwise be empty, like it doesn't make sense to rote a negative amount of times, and filling with random elements < 0, could be accomplished with Q.MUL -1.

If you feel it's messy, any chance you could suggest a split of this OP?

Reply:
> The comment was to remind the reader (you?) that Q.RND 0 doesn't do anything.
yeah and i think it's helpful, it was just the question marks that were confusing to me, as they imply this is an unresolved question.